test/api/v1/client_test.go

```diff
@@ -302,3 +311,77 @@ func TestClientJobMethods(t *testing.T) {
 	}, 100, 100*time.Millisecond)
 	require.NoError(t, err)
 }
+
+func TestClientUserMethods(t *testing.T) {
+	suite := test.NewTestSuite(t)
+	defer suite.Cleanup()
+
+	/////////////////////
+	// t.Run()
+	users, err := suite.APIClient.GetUsers(suite.Context(), nil)
+	require.NoError(t, err)
+	require.NotNil(t, users)
+	require.Empty(t, users.Users, "Expected no users in a fresh database")
+
+	t.Run("CreateUser_Success", func(t *testing.T) {
+		// Create first user
+		newUser1, err := suite.APIClient.CreateUser(suite.Context(), defaultUser1)
+		require.NoError(t, err)
+		require.NotEmpty(t, newUser1.UserId, "User ID should not be empty")
+
+		// Create first user
+		newUser2, err := suite.APIClient.CreateUser(suite.Context(), defaultUser2)
+		require.NoError(t, err)
+		require.NotEmpty(t, newUser2.UserId, "User ID should not be empty")
+	})
+
+	t.Run("CreateUser_DuplicateUsername", func(t *testing.T) {
+		// Try to create a user with the same username
+		duplicateUser := defaultUser1
+		_, err := suite.APIClient.CreateUser(suite.Context(), duplicateUser)
+		require.Error(t, err, "Creating user with duplicate username should fail")
+	})
+
+	t.Run("GetUserByID_Success", func(t *testing.T) {
+		// Create a user first
+		newUser, err := suite.APIClient.CreateUser(suite.Context(), infrastructure.CreateUserRequest{
+			Username:     "testuser_getbyid",
+			Email:        "getbyid@example.com",
+			Role:         1,
+			PublicSshKey: "ssh-rsa TESTKEY",
+		})
+		require.NoError(t, err)
+
+		// Get the user by ID
+		resp, err := suite.APIClient.GetUserByID(suite.Context(), fmt.Sprint(newUser.UserId))
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		require.Equal(t, "testuser_getbyid", resp.User.Username)
+		require.Equal(t, "getbyid@example.com", resp.User.Email)
+	})
+
+	t.Run("GetUserByUsername_Success", func(t *testing.T) {
+		// Create a user first
+		uniqueUsername := "unique_username_test"
+		_, err := suite.APIClient.CreateUser(suite.Context(), infrastructure.CreateUserRequest{
+			Username:     uniqueUsername,
+			Email:        "unique@example.com",
+			Role:         1,
+			PublicSshKey: "ssh-rsa UNIQUEKEY",
+		})
+		require.NoError(t, err)
+
+		// Get the user by username
+		userResp, err := suite.APIClient.GetUsers(suite.Context(), &models.UserQueryOptions{Username: uniqueUsername})
+		require.NoError(t, err)
+		require.NotNil(t, userResp.User)
+		require.Equal(t, uniqueUsername, userResp.User.Username)
+		require.Equal(t, "unique@example.com", userResp.User.Email)
+	})
+
+	t.Run("GetUserByUsername_NotFound", func(t *testing.T) {
+		// Try to get a non-existent username
+		_, err := suite.APIClient.GetUsers(suite.Context(), &models.UserQueryOptions{Username: "nonexistent_user"})
+		require.Error(t, err, "Getting non-existent username should return error")
+	})
```

Comment:
Nice, I would add one last test case here that gets all the users. Since the only time we hit that endpoint is in the beginning when no users exist. 
Doing it now at the end would ensure we pull all the users.